interfaces/apparmor/backend.go

```diff
@@ -295,6 +295,11 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	// Add snippets derived from the layout definition.
 	spec.(*Specification).AddSnapLayout(snapInfo)
 
+	if snapInfo.InstanceKey != "" {
```

Comment:
While it doesn't matter because actual assembly of the snippets is done in a fixed order could we please move that above the `AddSnapLayout` call? In addition we can move the `if snapInfo.InstanceKey` into the called method for resilience to mistakes and then make the call unconditional.